sdk/client-core/core/src/main/java/io/clientcore/core/util/ServerSentEventUtils.java

```diff
@@ -20,114 +22,121 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
-import java.util.function.Consumer;
 import java.util.regex.Pattern;
 
 /**
- * Utility class for Server Sent Event handling
+ * Utility class for Server Sent Event handling.
  */
-public final class ServerSentEventUtil {
+public final class ServerSentEventUtils {
     private static final String DEFAULT_EVENT = "message";
     private static final Pattern DIGITS_ONLY = Pattern.compile("^[\\d]*$");
     private static final String LAST_EVENT_ID = "Last-Event-Id";
     public static final String NO_LISTENER_ERROR_MESSAGE = "No ServerSentEventListener attached to HttpRequest to "
-            + "handle the text/event-stream response";
+        + "handle the text/event-stream response";
 
-    private ServerSentEventUtil() {
+    private ServerSentEventUtils() {
     }
 
     /**
-     * Checks if the content type is a text event stream
+     * Checks if the {@code Content-Type} is a text event stream.
      *
-     * @param contentType The content type
-     * @return True if the content type is a text event stream
+     * @param contentType The content type.
+     *
+     * @return {@code true} if the content type is a text event stream.
      */
     public static boolean isTextEventStreamContentType(String contentType) {
         if (contentType != null) {
-            return ContentType.TEXT_EVENT_STREAM.regionMatches(true, 0, contentType, 0, ContentType.TEXT_EVENT_STREAM.length());
+            return ContentType.TEXT_EVENT_STREAM.regionMatches(true, 0, contentType, 0,
+                ContentType.TEXT_EVENT_STREAM.length());
         } else {
             return false;
         }
     }
 
     /**
-     * Processes the text event stream
+     * Processes the text event stream.
      *
-     * @param httpRequest The HTTP Request
-     * @param httpRequestConsumer The HTTP Request consumer
-     * @param inputStream The input stream
-     * @param listener The listener object attached with the httpRequest
-     * @param logger The logger object
+     * @param httpClient The {@link HttpClient} to send the {@code httpRequest} through.
+     * @param httpRequest The {@link HttpRequest} to send.
+     * @param inputStream The {@link InputStream} to read data from.
+     * @param listener The {@link ServerSentEventListener event listener} attached to the {@link HttpRequest}.
      */
-    public static void processTextEventStream(HttpRequest httpRequest, Consumer<HttpRequest> httpRequestConsumer,
-        InputStream inputStream, ServerSentEventListener listener, ClientLogger logger) {
-        RetrySSEResult retrySSEResult;
+    public static void processTextEventStream(HttpClient httpClient, HttpRequest httpRequest,
```

Comment:
Yep, that was the reason. I don't think we should be passing around the client instance making it difficult in debuggability issues.

Reply:
How does passing an `HttpClient` instance negatively affect debuggability? I understand that users have finer control of how to handle a resulting `Response` by passing a `Consumer`, but if that's not something we want to allow to happen I think it makes for a worse API signature.

Reply:
Before it was an internal/implementation class. We didn't expose any specific loggers here. 
Also, on passing the actual httpClient to the user. We shouldn't be exposing more than whats needed. So exactly as you mentioned we only want users to get control of the ability to send the modified/redirected request instead of exposing the httpclient instance.

Reply:
Let's chat about this - I'm not entirely sure what the httpClient and httpRequest are being used for here.

Reply:
Based on our chat, I think lets leave this as-is and follow up in another PR with a cleanup of the SSE utils.